hail/python/hail/methods/family_methods.py

```diff
@@ -59,7 +60,55 @@ def trio_matrix(dataset, pedigree, complete_trios=False) -> MatrixTable:
     -------
     :class:`.MatrixTable`
     """
-    return MatrixTable._from_java(dataset._jmt.trioMatrix(pedigree._jrep, complete_trios))
+    mt = dataset
+    if not (len(mt.col_key) == 1 and mt.col_key[0].dtype == hl.tstr):
+        raise ValueError(f'trio_matrix requires column key of type [str], found: {",".join(list(mt.colkey))}')
+    
+    k = mt.col_key.dtype.fields[0]
+    samples = mt[k].collect()
```

Comment:
can just be `mt.col_key[0].collect()`

Reply:
I need `k` below.  I don't think changing the collect line is clearer.

Reply:
ah, right.